src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -392,11 +392,7 @@ public async Task<IActionResult> Edit(string id)
                 return Forbid();
             }
 
-            var shellSettings = _shellHost.GetAllSettings()
-                .Where(x => String.Equals(x.Name, id, StringComparison.OrdinalIgnoreCase))
-                .FirstOrDefault();
-
-            if (shellSettings == null)
+            if (!_shellHost.TryGetSettings(id, out var shellSettings))
```

Comment:
Yes

That's funny, I already did the same in my PR #11890 for tenant removing, I did it because anyway there are many places where it is intended to pass the exact shell settings Name. By changing the inner dicos to be case insensitive we retrieve somewhere the old behavior where it was already not allowed to create 2 "Foo" and "foo" tenants.

You are breaking my PR ;)

Reply:
I see. You change will break existing code since it no longer check for invariant case. I think this change should be made to this PR since it has the complete fix. Let me know if you want me to recent this change anyway.

Reply:
First yes, you can do these kind of changes here, no problem.

> I think the change in the other PR will break existing code

It was intentional as in some place the passed tenant name is intended to be the exact tenant name, e.g the Tenants view generate action link based on existing shellSettings.Names, idem for the openId where tenants has been selected from existing shellSettings.Names.

Mainly I mean we only need to update the places where we were doing ignore case lookup which is now useless, but not the opposite, hmm I think we only need to update the tenants admin controller.

Reply:
@jtkech can you kindly review the code change one more time? I reverted some of the original changes as per your request.

Reply:
No problem, will do it tomorrow.